From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@somewhere.on.planet>
Date: Mon, 19 Feb 2024 02:14:58 +0000
Subject: Patching kernel sunxi files include/uapi/drm/sun4i_drm.h

Signed-off-by: John Doe <john.doe@somewhere.on.planet>
---
 include/uapi/drm/sun4i_drm.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/uapi/drm/sun4i_drm.h b/include/uapi/drm/sun4i_drm.h
index eb9d314893ef..e0783a398481 100644
--- a/include/uapi/drm/sun4i_drm.h
+++ b/include/uapi/drm/sun4i_drm.h
@@ -11,8 +11,12 @@
 #ifndef __SUN4I_DRM_H__
 #define __SUN4I_DRM_H__
 
 #include "drm.h"
 
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
 struct drm_sun4i_gem_create {
 	__u64 size;
 	__u32 flags;
@@ -28,6 +32,10 @@ struct drm_sun4i_gem_create {
 #define DRM_SUN4I_GEM_CREATE		0x00
 
 #define DRM_IOCTL_SUN4I_GEM_CREATE	DRM_IOWR(DRM_COMMAND_BASE + DRM_SUN4I_GEM_CREATE, \
 						 struct drm_sun4i_gem_create)
 
+#if defined(__cplusplus)
+}
+#endif
+
 #endif
-- 
Created with Armbian build tools https://github.com/armbian/build

